### DISCUSSION TITLE
Skip validation until the first form submit?

### DISCUSSION THREAD

**Discussion by jaredcwhite** [pablo-abc/felte#129] [Ideas]
Awesome library BTW! One question: I noticed the validation happens on fields immediately after a blur on the field, even if you haven't submitted the form yet. I'd like to have behavior where you only see validation errors upon submitting the form (and then it can check in real-time after that as you interact on fields). Is there any way to enable that, or if not…something you'd be able to add?

**Comment by pablo-abc** [comment]
Felte has been growing gradually based on user requirements. While I know this behaviour exists in libraries such as React Hook Form, the most common use case seems to be validation on touch + on change after it's been touched. So currently this is the only supported behaviour.

> Sidenote: `touched` refers to how Felte defines when a user has interacted with the field. A field is. touched on blur on text related fields, and on change on others (such as a checkbox).

Adding this behaviour is possible, though! The expected behaviour would be validation on submit, and subsequently on change, right? I might find some time during the weekend to work on this. Feel free to open an issue to track this, though!

**Reply by jaredcwhite** [reply]
I mean, it's certainly no showstopper for me, and in some scenarios I think the immediate feedback would actually be helpful. But I can imagine it would be helpful to have that option for more control over the UX.

**Reply by pixelmund** [reply]
Was looking in the docs for a while and finally found this discussion, it would be nice to have this configurable!
